File: src/types/song.ts
export type SongId = string;

export type Song = {
  channel: Channel;
  duration: Duration;
  id: SongId;
  name: string;
  published: string;
  thumbnail: string;
  views: Views;
};

export type Channel = {
  name: string;
  url: string;
};

export type Duration = {
  label: string;
  seconds: number;
};

export type Views = {
  count: number;
  label: string;
};

File: src/models/song.go
package models

type Song struct {
	Channel *struct {
		Name string `json:"name"`
		Url  string `json:"url"`
	} `json:"channel"`

	Duration *struct {
		Count uint32 `json:"count"`
		Label string `json:"label"`
	} `json:"duration"`

	Id        string `json:"id"`
	Name      string `json:"name"`
	Published string `json:"published"`
	Thumbnail string `json:"thumbnail"` // might be nullable

	Views *struct {
		Count uint32 `json:"count"`
		Label string `json:"label"`
	} `json:"views"`
}
